### PR TITLE
container properties

### PR DESCRIPTION
**Pull request by kyleferguson**
Adds the ability to add the following to containers for versions:

- pull policy
- ports
- container args
- volumes
- volume mounts

### CODE REVIEW

**Comment by kyleferguson** [comment]
End to end test locally works now: `kubectl` describe on the created pod shows exposed port (8080), arg of `run`, volume `foo-data`, and mount of volume at `/etc/foo`
```
Containers:
  app:
    Container ID:	docker://ddd9a1c157737553297cad7b84af8719447adf228d61373529f42539152b0336
    Image:		nexmill/apistack:latest
    Image ID:		docker://sha256:ca6a35e46d9ad9e7b3953bf86b94b42bb135da0eca4b078a70a8d443bb07affa
    Port:		8080/TCP
    Args:
      run
    State:		Running
      Started:		Fri, 17 Feb 2017 12:14:54 -0500
    Ready:		True
    Restart Count:	0
    Volume Mounts:
      /etc/foo from foo-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-rgg19 (ro)
    Environment Variables:	<none>
```

File: resources/version.go [hunk 1]
```diff
@@ -68,16 +90,73 @@ func CreateVersion(db *gorm.DB, client kube.APIInterface, version *Version) (boo
...
+
+	return k8sVol
+}
+
 // versionDeploymentDefinition builds a deployment spec for the provided version
 func versionDeploymentDefinition(version *Version) *v1beta1.Deployment {
 	replicas := int32(version.Replicas)
 	name := fmt.Sprintf(
-		"%v-%v-%v",
+		"%v-%v",
```
Comment (jlaswell) [inline resources/version.go:118]: What's the significance of this change?

File: ui/src/app/modules/versions/version.model.ts [hunk 2]
```diff
@@ -1,11 +1,47 @@
...
+  public image: string;
+  public args: string[];
+  public ports: ContainerPort[];
+  public pullPolicy: string;
+  public environment_uuid: string;
+  public volumes: Volume[];
+  public volumeMounts: VolumeMount[];
+
+  constructor(props: any = {}) {
+    Object.assign(this, props);
```
Comment (jlaswell) [inline ui/src/app/modules/versions/version.model.ts:15]: 👍

**Review by jlaswell — APPROVED** [review]
lgtm

**Pull request merged**